adminapi/src/main/java/io/minio/admin/MinioAdminClient.java

```diff
@@ -613,6 +620,241 @@ public Message getServerInfo() throws IOException, NoSuchAlgorithmException, Inv
     }
   }
 
+  /**
+   * Creates a new service account belonging to the user sending.
+   *
+   * <pre>Example:{@code
+   * // Assume policyJson contains below JSON string;
+   * // {
+   * //     "Statement": [
+   * //         {
+   * //             "Action": "s3:GetObject",
+   * //             "Effect": "Allow",
+   * //             "Principal": "*",
+   * //             "Resource": "arn:aws:s3:::my-bucketname/myobject*"
+   * //         }
+   * //     ],
+   * //     "Version": "2012-10-17"
+   * // }
+   * //
+   * }</pre>
+   *
+   * @param accessKey Access key.
+   * @param targetUser Target user.
+   * @param secretKey Secret key.
+   * @param policy Policy as JSON string .
+   * @param name Service account name.
+   * @param expiryTime Expiry time , Example : 2023-12-02T15:04:05Z.
+   * @throws NoSuchAlgorithmException thrown to indicate missing of MD5 or SHA-256 digest library.
+   * @throws InvalidKeyException thrown to indicate missing of HMAC SHA-256 library.
+   * @throws IOException thrown to indicate I/O error on MinIO REST operation.
+   * @throws InvalidCipherTextException thrown to indicate data cannot be encrypted/decrypted.
+   */
+  public AddServiceAccountReq addServiceAccount(
+      @Nonnull String targetUser,
+      @Nullable String name,
+      @Nullable String secretKey,
+      @Nullable String accessKey,
+      @Nullable String policy,
+      @Nullable String description,
+      @Nullable String expiryTime)
+      throws NoSuchAlgorithmException, InvalidKeyException, IOException,
+          InvalidCipherTextException {
+    if (targetUser == null || targetUser.isEmpty()) {
+      throw new IllegalArgumentException("target user must be provided");
+    } else if (targetUser.length() > 32) {
+      throw new IllegalArgumentException("name must not be longer than 32 characters");
+    }
+    if (description != null && description.length() > 256) {
+      throw new IllegalArgumentException("description must be at most 256 bytes long");
+    }
+    if (accessKey == null || accessKey.isEmpty()) {
+      accessKey = generateCredentials(20);
+    }
+    if (secretKey == null || secretKey.isEmpty()) {
+      secretKey = generateCredentials(40);
+    }
+    byte[] policyBytes = null;
+    if (policy != null && !policy.isEmpty()) {
+      policyBytes = policy.getBytes(StandardCharsets.UTF_8);
+    }
+
+    Credentials creds = getCredentials();
+    AddServiceAccountReq addServiceAccountReq =
+        new AddServiceAccountReq(
+            secretKey, policyBytes, targetUser, accessKey, name, description, expiryTime);
```

Comment:
You could use `HashMap()` and convert to JSON string directly instead of intermittent `AddServiceAccountReq` class.